tests/resources/main.tf

```diff
@@ -11,3 +11,69 @@ module "landing_zone" {
   enable_transit_gateway = false
   add_atracker_route     = false
 }
+
+################################################################################
+# Resource Group
+################################################################################
+module "resource_group" {
+  source              = "terraform-ibm-modules/resource-group/ibm"
+  version             = "1.1.6"
+  resource_group_name = "${var.prefix}-rg"
+}
+
+module "existing_sm_crn_parser" {
+  # count   = var.existing_secrets_manager_instance_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.0.0"
+  crn     = var.existing_secrets_manager_instance_crn
+}
+
+#################################################################################
+# Secrets Manager resources
+#################################################################################
+
+# Create a secret group to place the certificate if provisioning a new certificate
+module "secrets_manager_group" {
+  source                   = "terraform-ibm-modules/secrets-manager-secret-group/ibm"
+  version                  = "1.2.2"
+  region                   = module.existing_sm_crn_parser.region
+  secrets_manager_guid     = module.existing_sm_crn_parser.service_instance
+  secret_group_name        = "${var.prefix}-cert-secret-group"
+  secret_group_description = "secret group used for private certificates"
+  providers = {
+    ibm = ibm.ibm-sm
+  }
+}
+
+# Configure private cert engine if provisioning a new certificate
+module "private_secret_engine" {
```

Comment:
Oh don't try to deploy a new engine in our permanent SM instance! There is already an engine installed! 
So we either use existing SM and engine, or have this test create new SM instance and engine

Reply:
for existing test: existing SM instance and engine is used
for standard test in schematics: new SM instance is created